annotations/annotations_rw.go

```diff
@@ -66,7 +68,49 @@ func (rw *annotationsRW) Read(ctx context.Context, contentUUID string) ([]*Annot
 	case http.StatusNotFound:
 		return nil, false, nil
 	default:
-		return nil, false, fmt.Errorf("annotations RW returned an unexpected HTTP status code: %v", resp.StatusCode)
+		return nil, false, fmt.Errorf("annotations RW returned an unexpected HTTP status code in read operation: %v", resp.StatusCode)
+	}
+}
+
+func (rw *annotationsRW) Write(ctx context.Context, contentUUID string, annotations []Annotation) error {
+	tid, err := tidUtils.GetTransactionIDFromContext(ctx)
+
+	if err != nil {
```

Comment:
I wonder whether this checking should be moved to the resource. It's quite ugly and repetitive and I think the resource should be responsible for preparing the context that's propagated to the service.

Reply:
Based on the current code, this condition is never `true` since the handler always generates a valid context even if the the transaction ID header is missing in the requests.

My personal problem with Go is that I can not sleep at night if I do not handle all the possible errors. :P

I am happy to drop it if it improves readability of the code.

Reply:
It can stay!